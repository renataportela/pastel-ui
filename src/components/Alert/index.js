export { default, ALERT_COLORS } from './Alert'