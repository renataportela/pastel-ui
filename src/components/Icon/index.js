export { default, iconNames } from './Icon';
export { default as Svg } from './Svg';