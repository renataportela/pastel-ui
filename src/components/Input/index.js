export { default, INPUT_PROPS } from './Input'
export { inputStyles } from './ElementStyle'
