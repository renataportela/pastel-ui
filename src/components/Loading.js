import React from 'react'
import { colors, Icon, Overlay } from '~/pastel-ui'

function Loading(props){
  return (
    <Overlay>
      <Icon name="loading" width={30} fill={props.color || colors.secondary} />
    </Overlay>
  )
}

export default Loading