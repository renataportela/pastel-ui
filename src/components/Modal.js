import React, { useEffect, useRef, useState } from 'react';
import ReactDOM from 'react-dom';
import styled from '@emotion/styled';

import { scale } from '~/pastel-ui/theming';
import { Overlay, Paper } from '~/pastel-ui';
import Button from '~/pastel-ui/components/Button';
import useOutsideCloser from '~/pastel-ui/hooks/useOutsideCloser';

const modalRoot = document.getElementById('modal-root');

const Centered = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 300;

  & > div {
    min-width: 300px;
    max-width: 90%;
    animation: ${scale} .1s;

    @media (min-width: 778px) {
      min-width: 600px;
    }
  }
`;

const CloseContainer = styled.div`
  float: right;
  margin-left: 20px;
  margin-bottom: 20px;
`;

const CloseButton = styled(Button)`
  position: absolute;
  top: 12px;
  right: 12px;
`;

function Modal(props){
  const [el] = useState(document.createElement('div'));
  const modalBoxRef = useRef();
  const close = () => props.setOpen(false);

  useOutsideCloser(modalBoxRef, close);

  useEffect(() => {
    modalRoot.appendChild(el);
    return () => modalRoot.removeChild(el);
  }, []);  

  if (props.open){
    return ReactDOM.createPortal(
        <Overlay>
          <Centered>
            <Paper ref={modalBoxRef}>
              <CloseContainer>
                <CloseButton 
                  pallete="primary" 
                  variant="outline" 
                  title={props.titleClose || 'Close'}
                  icon="close"
                  round
                  onClick={close}
                />
              </CloseContainer>
              {props.children}
            </Paper>
          </Centered>
        </Overlay>,
      el
    );
  }

  return null;
}

export default Modal;