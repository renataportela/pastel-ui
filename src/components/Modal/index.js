export { default } from './Modal'
export { default as ModalBase } from './ModalBase'