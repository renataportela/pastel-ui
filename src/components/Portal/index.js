export { default } from './Portal'
export { prepareClientPortals } from '@jesstelford/react-portal-universal'