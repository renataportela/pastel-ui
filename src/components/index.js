// Content
export { default as Avatar } from './content/Avatar'
export { default as Badge } from './content/Badge'
export { default as Chip } from './content/Chip'
export { default as Icon, iconNames, Svg } from './content/Icon'
export { default as Tag } from './content/Tag'
export { Heading, Paragraph, Text } from './content/typography'

// Data Entry
export { default as TextField } from './dataEntry/TextField'

// Dialog
export { default as Alert } from './dialog/Alert'

// Layout
export * from './layout/Grid'
export { default as BackTop } from './layout/BackTop'
export { default as Button, Close } from './layout/Button'
export { default as Ripple } from './layout/Ripple'
export { default as Theme, BaseColors, useTheme } from './layout/Theme'