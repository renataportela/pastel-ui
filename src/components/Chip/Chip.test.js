import React from 'react'
import renderer from 'react-test-renderer'

import { Theme, Chip } from '~/components'

describe('Chip component', () => {
  it('renders a chip with default color and close button', () => {
    const tree = renderer
      .create(<Theme><Chip onDelete={() => {}}>Default</Chip></Theme>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a chip with an Avatar image', () => {
    const tree = renderer
      .create(
        <Theme>
          <Chip avatar={{ image: 'https://ps.w.org/wp-avatar/assets/icon-256x256.png' }}>
            Avatar
          </Chip>
        </Theme>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a chip with warning color', () => {
    const tree = renderer
      .create(
        <Theme>
          <Chip color="warning">Warning</Chip>
        </Theme>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})