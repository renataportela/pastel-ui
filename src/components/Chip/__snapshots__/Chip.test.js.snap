// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip component renders a chip with an Avatar image 1`] = `
<div
  className="BaseColors__BaseStyle-s1fbie-0 dWoZEV Chip__ChipStyle-sc-8ucx7t-0 fWpUtP"
>
  <span
    className="BaseColors__BaseStyle-s1fbie-0 cKbdWp Avatar__Round-sc-17gj85m-0 bjPCmA"
    size="sm"
  >
    <img
      className="Avatar__Image-sc-17gj85m-1 ftoiXU"
      src="https://ps.w.org/wp-avatar/assets/icon-256x256.png"
    />
    <span
      className="Avatar__Text-sc-17gj85m-2 cFsSmv"
    />
  </span>
  <span
    className="Chip__Content-sc-8ucx7t-1 hGJwQy"
  >
    Avatar
  </span>
</div>
`;

exports[`Chip component renders a chip with default color and close button 1`] = `
<div
  className="BaseColors__BaseStyle-s1fbie-0 dWoZEV Chip__ChipStyle-sc-8ucx7t-0 fWpUtP"
>
  <span
    className="Chip__Content-sc-8ucx7t-1 hGJwQy"
  >
    Default
  </span>
  <button
    className="ButtonStyle-sc-120zjl4-0 anqYW"
    color="sub"
    disabled={false}
    kind="ghost"
    size="sm"
  >
    <svg
      className="Svg__SvgStyle-pyey0i-0 gYARyh"
      fill="none"
      height="24px"
      size={null}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={2}
      viewBox="0 0 24 24"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="18"
        x2="6"
        y1="6"
        y2="18"
      />
      <line
        x1="6"
        x2="18"
        y1="6"
        y2="18"
      />
    </svg>
  </button>
</div>
`;

exports[`Chip component renders a chip with warning color 1`] = `
<div
  className="BaseColors__BaseStyle-s1fbie-0 jTDUIl Chip__ChipStyle-sc-8ucx7t-0 fWpUtP"
>
  <span
    className="Chip__Content-sc-8ucx7t-1 hGJwQy"
  >
    Warning
  </span>
</div>
`;
